files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,442 @@ class AaaCfg(object):
+CFG_MGMT_VRF_KEY  = "vrf_global"
+MGMT_VRF_TABLE_ID = 5000
+
+class MgmtVrfCfg:
+    def __init__(self, config_db):
+        syslog.syslog(syslog.LOG_INFO, 'Initializing MGMT VRF handler')
+        self.config_db = config_db
+        self.mgmt_vrf_table_created = False
+
+    def create_mgmt_vrf_table(self):
+        if self.mgmt_vrf_table_created == False:
+            syslog.syslog(syslog.LOG_INFO, "Create MGMT VRF table:")
+            cmd = 'ip link add name mgmt type vrf table {}'.format(MGMT_VRF_TABLE_ID)
+            if run_command(cmd) == True:
+                self.mgmt_vrf_table_created = True
+
+    def delete_mgmt_vrf_table(self):
+        if self.mgmt_vrf_table_created == True:
+            syslog.syslog(syslog.LOG_INFO, "Delete MGMT VRF table:")
+            cmd = 'ip link delete mgmt'
+            if run_command(cmd) == True:
+                self.mgmt_vrf_table_created = False
+
+    def configure_mgmt_vrf(self):
+        syslog.syslog(syslog.LOG_INFO, "Configure MGMT VRF:")
+
+        self.create_mgmt_vrf_table()
+
+        cmd = 'ip link set dev mgmt up'
+        if run_command(cmd) == False:
+            return
+
+        syslog.syslog(syslog.LOG_INFO, "Add lo intf:")
+        cmd = 'ip link add lo-m type dummy'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip addr add 127.0.0.1/8 dev lo-m'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link set lo-m up'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link set dev lo-m master mgmt'
+        if run_command(cmd) == False:
+            return
+
+        self.bind_mgmt_intf()
+
+    def bind_mgmt_intf(self):
+        syslog.syslog(syslog.LOG_INFO, "Handle MGMT intf:")
+        keys = self.config_db.get_keys(CFG_MGMT_INTF_TABLE)
+        syslog.syslog(syslog.LOG_INFO, "MGMT interfaces keys: {}".format(str(keys)))
+
+        mgmt_intf_bound = False
+        for it in keys:
+            if type(it) is unicode:
+                mgmt_intf_name = it
+
+                cmd = 'ip link set dev {} master mgmt'.format(mgmt_intf_name)
+                if run_command(cmd) == False:
+                    return
+
+                cmd = 'sysctl -w net.ipv4.tcp_l3mdev_accept=1'
+                if run_command(cmd) == False:
+                    return
+
+                mgmt_intf_bound = True
+                break
+        if not mgmt_intf_bound:
+            syslog.syslog(syslog.LOG_ERR, "MGMT VRF: MGMT interface not bound")
+        else:
+            for it in keys:
+                if type(it) is unicode:
+                    key = it
+                else:
+                    key = it[0] + "|" + it[1]
+
+                entry = self.config_db.get_entry(CFG_MGMT_INTF_TABLE, key)
+                if 'gwaddr' in entry:
+                    mgmt_intf_gwaddr = entry['gwaddr']
+                    cmd = 'ip route add table {} 0.0.0.0/0 via {}'.format(MGMT_VRF_TABLE_ID, mgmt_intf_gwaddr)
```

Comment:
You could reuse the same as in handle_mgmt_intf_gwaddr_cfg

Reply:
done